lib/fluent/plugin/in_opensearch.rb

```diff
@@ -321,7 +374,17 @@ def run_slice(slice_id=nil)
       end
 
       router.emit_stream(@tag, es)
+      clear_scroll(scroll_id)
+      #reset steps and next_time if our function successful ends
+      @retry.instance_variable_set(:@steps, 0)
+      @retry.instance_variable_set(:@next_time, nil)
```

Comment:
Basically, your patch is almost overwhelmed my expectations! Quite good work.
One thing, retry_state instance should be reinitialized like as plugin/output.rb does in Fluentd core:
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/output.rb#L1119
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/output.rb#L1314-L1315

So, eliminate the contents of `@retry` here with: `@retry = nil`
And, recreating in https://github.com/fluent/fluent-plugin-opensearch/pull/136/files#diff-61847321aa21e1cfdbc79e273831965c71d90b38ed24e8fe480bf5bcc908e0e4R351 with

```ruby
unless @retry
   @retry = retry_state(@retry_randomize)
end
```
would be needed. Other places are perfectly implemented. :+1: